plugins/providers/docker/driver.rb

```diff
@@ -24,13 +24,13 @@ def build(dir, **opts, &block)
         args << dir
         opts = {with_stderr: true}
         result = execute('docker', 'build', *args, opts, &block)
-        matches = result.match(/Successfully built (?<id>.+)$/i)
+        matches = result.scan(/Successfully built (?<id>.+)$/i).last
         if !matches
           # Check for the new output format 'writing image sha256...'
           # In this case, docker builtkit is enabled. Its format is different
           # from standard docker
           @logger.warn("Could not determine docker container ID. Scanning for buildkit output instead")
-          matches = result.match(/writing image .+:(?<id>[0-9a-z]+) done/i)
+          matches = result.scan(/writing image .+:(?<id>[0-9a-z]+) done/i).last
```

Comment:
Since the conversion to `#scan` to replace `#match` doesn't use the match identifier (`?<id>`) it can be removed from the regex here.